Firestore/Protos/nanopb_cpp_generator.py

```diff
@@ -246,16 +271,149 @@ def nanopb_write(results):
-    f.name = result['sourcename']
-    f.content = result['sourcedata']
+    source_request = FileGenerationRequest(response.file, result['sourcename'],
+                                           nanopb_fixup(result['sourcedata']))
+    nanopb_augment_source(source_request, pretty_printing_generator)
 
   return response
 
 
+class FileGenerationRequest:
+  """Represents a request to generate a file.
+
+  The initial contents of the file can be augmented by inserting extra text at
+  insertion points. For each file, Nanopb defines the following insertion points:
+
+  - 'includes' -- beginning of the file, after the last Nanopb include;
+  - 'eof' -- the very end of file, right before the include guard.
+
+  In addition, each header also defines a 'struct:Foo' insertion point inside
+  each struct declaration, where 'Foo' is the the name of the struct.
+
+  See the official protobuf docs for more information on insertion points:
+  https://github.com/protocolbuffers/protobuf/blob/129a7c875fc89309a2ab2fbbc940268bbf42b024/src/google/protobuf/compiler/plugin.proto#L125-L162
+  """
+
+  def __init__(self, files, file_name, contents):
+    """
+    Args:
+      files: The array of files to generate inside a `CodeGenerationRequest`.
+        New files will be added to it.
+      file_name: The name of the file to generate/augment.
+      contents: The initial contents of the file, before any augmentation, as
+        a single string.
+
+    """
+    self.files = files
+    self.file_name = file_name
+
+    self._set_contents(contents)
+
+  def _set_contents(self, contents):
+    """Creates a request to generate a new file with the given `contents`.
+    """
+    f = self.files.add()
+    f.name = self.file_name
+    f.content = contents
+
+  def insert(self, insertion_point, to_insert):
+    """Adds extra text to the generated file at the given `insertion_point`.
+
+    Args:
+      insertion_point: The string identifier of the insertion point, e.g. 'eof'.
+        The extra text will be inserted right before the insertion point. If
+        `insert` is called repeatedly, insertions will be added in the order of
+        the calls.
+      to_insert: The text to insert as a string.
+    """
+    f = self.files.add()
+    f.name = self.file_name
+    f.insertion_point = insertion_point
+    f.content = to_insert
+
+
+def nanopb_fixup(file_contents):
+  """Applies fixups to generated Nanopb code.
+
+  This is for changes to the code, as well as additions that cannot be made via
+  insertion points. Current fixups:
+  - rename fields named `delete` to `delete_`, because it's a keyword in C++.
+
+  Args:
+    file_contents: The contents of the generated file as a single string. The
+      fixups will be applied without distinguishing between the code and the
+      comments.
+  """
+
+  delete_keyword = re.compile(r'\bdelete\b')
+  return delete_keyword.sub('delete_', file_contents)
+
+
+def nanopb_augment_header(request, pretty_printing_generator):
```

Comment:
The `pretty_printing_generator` name suggests that it might work with any generator, when it really is the file one.

Consider renaming to `file_generator` to make the parameter name more clearly indicate the restriction.

Reply:
Renamed to `file_pretty_printing`.